osdk/tools/Dockerfile.tdx

```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: MPL-2.0
+
+# This image is for the OSDK GitHub CI. 
+# The environment is consistent with the one 
+# described in the OSDK User Guide section of the Asterinas Book.
+#
+# TODO: We should build the Asterinas image based on the OSDK image
+# since Asterinas is managed by OSDK itself.
+# However, currently, these two images have different contents.
+# The main distinction is that
+# QEMU, grub, and OVMF in the OSDK image are installed via apt,
+# while these tools in the Asterinas image are built from source.
+# Some boot methods in Asterinas only function properly
+# when using the tools that are built from source.
+
+FROM intelcczoo/tdvm:ubuntu22.04-mvp_2023ww15 as build-base
```

Comment:
This seems to be the single line that is different from `Dockerfile.ubuntu22.04`. We need to find a way to avoid having the two almost identical Dockefiles, which is not maintainable.

Consider adopting a simple template-based scheme by adding a Dockerfile generator script named `gen_dockerfile.sh`

```
Overview: The Dockerfile generator for OSDK.

Usage: gen_dockerfile.sh [OPTIONS]

Options:
  --help          Print the help information
  --intel-tdx     Include Intel TDX support
  --out-dir       Output the Dockerfile under this directory. By default, the
                  output directory is the current working directory.
```

The Dockerfile template can be embedded inside the script. Running the script generates the Dockefile according to the given CLI arguments.